src/main/java/net/sf/jabref/logic/integrity/ISBNChecker.java

```diff
@@ -0,0 +1,69 @@
+package net.sf.jabref.logic.integrity;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.logic.integrity.IntegrityCheck.Checker;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class ISBNChecker implements Checker {
+
+    private static final Pattern ISBN_PATTERN = Pattern.compile("^(\\d{9}[\\dxX]|\\d{13})$");
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        if (!entry.hasField("isbn")) {
+            return Collections.emptyList();
+        }
+
+        // Check that the ISSN is on the correct form
+        String isbn = entry.getFieldOptional("isbn").get().trim().replace("-", "");
+        Matcher isbnMatcher = ISBN_PATTERN.matcher(isbn);
```

Comment:
Again, how to handle invalid data? Here it is slightly more obvious as the dashes are ignored, but should one store both version? Some people may prefer the dashes in particular positions...

I do not see this happening in this PR.